autoscheduler/scraper/tests/api_tests.py

```diff
@@ -525,3 +577,142 @@ def test_api_course_search_serializer_get_course_formats_correctly(self):
+                        "S": 0,
+                        "U": 0,
+                        "Q": 2,
+                        "X": 0
+                    }
+                   }
+
+        # Act
+        serializer = GradeSerializer(
+            Grades.objects.instructor_performance('ASCC', 101, 'Morgan W. Jones'))
+        # Assert
+        self.assertEqual(expected, serializer.data)
+
+    def test_api_grade_serializer_gives_expected_output_one_section(self):
+        """ Tests that the grades serializer returns the correct data
+            for one section """
+        # Arrange
+        expected = {'grades':
+                    {
+                        "gpa": 2.893,
+                        "A": 17,
+                        "B": 21,
+                        "C": 14,
+                        "D": 3,
+                        "F": 1,
+                        "I": 0,
+                        "S": 0,
+                        "U": 0,
+                        "Q": 2,
+                        "X": 0
+                    }
+                   }
+
+        # Act
+        serializer = GradeSerializer(
+            Grades.objects.instructor_performance('CSCE', 310, 'Ronald Ward'))
+        # Assert
+        self.assertEqual(expected, serializer.data)
+
+    def test_api_grades_gives_valid_response_ascc_201_jones(self):
+        """ Tests that /api/grades?subject=ASCC&course_num=101&
+            instructor=Morgan W. Jones gives the correct output
+        """
+        # Arrange
+        expected = {'grades':
+                    {
+                        "gpa": 3.3434333333333335,
+                        "A": 23,
+                        "B": 22,
+                        "C": 4,
+                        "D": 1,
+                        "F": 0,
+                        "I": 0,
+                        "S": 0,
+                        "U": 0,
+                        "Q": 2,
+                        "X": 0
+                    }
+                   }
+        data = {'subject': 'ASCC', 'course_num': '101', 'instructor': 'Morgan W. Jones'}
+
+        # Act
+        response = self.client.get('/api/grades', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected)
+
+    def test_api_grades_gives_valid_response_bims_110_young(self):
+        """ Tests that /api/grades?subject=BIMS&course_num=110&
+            instructor=Colin Young gives the correct output
+        """
+        # Arrange
+        expected = {'grades':
+                    {
+                        "gpa": 3.4265,
+                        "A": 193,
+                        "B": 6,
+                        "C": 0,
+                        "D": 0,
+                        "F": 0,
+                        "I": 0,
+                        "S": 0,
+                        "U": 0,
+                        "Q": 0,
+                        "X": 0
+                    }
+                   }
```

Comment:
If we change this to use an `APIView` instead of serializing a dict, we don't need to nest it inside of the `grades` key
```suggestion
        expected = {
                        "gpa": 3.4265,
                        "A": 193,
                        "B": 6,
                        "C": 0,
                        "D": 0,
                        "F": 0,
                        "I": 0,
                        "S": 0,
                        "U": 0,
                        "Q": 0,
                        "X": 
        }
```